StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp7/SpacingRules/SA1000CSharp7UnitTests.cs

```diff
@@ -228,15 +231,16 @@ public void TestMethod()
 }
 ";
 
-            DiagnosticResult[] expectedDiagnostics =
+            await new CSharpTest
             {
-                this.CSharpDiagnostic().WithArguments("foreach", string.Empty, "followed").WithLocation(7, 13),
-                this.CSharpDiagnostic().WithArguments("foreach", string.Empty, "followed").WithLocation(8, 13),
-            };
-
-            await this.VerifyCSharpDiagnosticAsync(testCode, expectedDiagnostics, CancellationToken.None).ConfigureAwait(false);
-            await this.VerifyCSharpDiagnosticAsync(fixedCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
-            await this.VerifyCSharpFixAsync(testCode, fixedCode).ConfigureAwait(false);
+                TestCode = testCode,
```

Comment:
I prefer ctor arguments and get only properties for required stuff.

Reply:
I tend to agree, but I like the uniformity of this approach when `TestCode` is the only required property. In the end, I'll probably have a helper method available to shorten the two primary and simplest use cases:

1. Verify code does not report diagnostics
2. Verify code reports diagnostics, and the fixer fixes all of them

The longer form seen here becomes only necessary for more complicated cases.